### DISCUSSION TITLE
Hubsite ID as Result source ID

### DISCUSSION THREAD

**Discussion by AkiraSPO** [microsoft-search/pnp-modern-search#2372] [Q&A] [answered]
Can I use a site GUID or a HubSite GUID as Result source id in pnp search webpart? Or do I have to create a result source using 'Manage Result Sources' page first to limit the search results?

I want to configure my search results webpart to search content only within a SPO hub.

**Comment by FranckyC** [comment, marked as answer]
The result source ID is not meant to be a site ID nor a hub ID but for the result source ID created from the SharePoint search settings (site collection or tenant). If you want to restrict results for the current hubsite, you can use the query template `DepartmentId:\{{Hub.HubSiteId}\}` for instance.

See:
- https://microsoft-search.github.io/pnp-modern-search/usage/search-results/tokens/#site-web-hub-etc-tokens
- https://github.com/microsoft-search/pnp-modern-search/discussions/1907
- https://www.techmikael.com/2018/04/working-with-hub-sites-and-search-api.html
